cloud/main.js

```diff
@@ -84,28 +86,37 @@ function sendToFeedbackChannel(body, response) {
 //   "endTimeToken": "00000"}
 
 Parse.Cloud.define("replayEvents", function(request, response) {
-    var requiredParams = ["aliasId", "subkey"];
+    var requiredParams = ["aliasId"];
     var params = request.params;
     checkMissingParams(params, requiredParams, response);
 
-    var count = params.count ? params.count : 9999;
-    var aliasId = params.aliasId;
+    var count = params.count || 20;
     var subkey = params.subkey;
+    var startTimeToken = params.startTimeToken ?
+        new Moment(params.startTimeToken).toDate() : new Moment().toDate();
+    var endTimeToken = params.endTimeToken ? new Moment(params.endTimeToken).toDate() : undefined;
+    var aliasId = params.aliasId;
 
-    Alias.get(params.aliasId).then(function(alias) {
-        var chatRoomId = alias.get("chatRoomId");
-        var startTimeToken = params.startTimeToken
-            ? params.startTimeToken
-            : new Date().getTime() * 10000;
-        var endTimeToken = params.endTimeToken;
-
-        Pubnub.replayChannel({subkey: subkey,
-                              channel: chatRoomId,
-                              startTimeToken: startTimeToken,
-                              endTimeToken: endTimeToken,
-                              count: count
-                             }).then(response.success, response.error);
-    }, response.error);
+    ChatEvent.getChatEvents(aliasId, startTimeToken, count, endTimeToken).then(function(chatEvents) {
+        var startTimeToken = chatEvents.length > 0 ?
```

Comment:
how would you do this? can you declare a var in an if statement and use it outside that block?
```
if(cond) {
  var x = 1
}
x + 2```